lib/models/api_calls.dart

```diff
@@ -98,5 +110,108 @@ class ApiCalls {
     }
   }
 
-  // TODO Remote commands Issue #38
+  /// Calls the Directory API to deprovise a device.
+  ///
+  /// * [authToken] is the authentication token for API to accept the requests.
+  /// * [deviceId] is the ID of the device which its detailed information
+  /// will be retrieved.
+  /// * [reason] is the reason why the device is getting a deprovision.
+  ///
+  /// This method returns true if the action was completed.
+  dynamic deprovision(String authToken, String deviceId, int reason) async {
+    var response;
+    authToken = 'Bearer $authToken';
+    _client == null
+        ? response = await http.post(
+            PREFIX + DETAILED_DEVICE_URL + deviceId + ACTION_URL,
+            headers: {HttpHeaders.authorizationHeader: authToken},
+            body: jsonEncode(<String, String>{
+              'action': 'deprovision',
+              'deprovisionReason': deprovisionReasons[reason]
+            }),
+          )
+        : response = await _client.post(
+            PREFIX + DETAILED_DEVICE_URL + deviceId + ACTION_URL,
+            headers: {HttpHeaders.authorizationHeader: authToken},
+            body: jsonEncode(<String, String>{
+              'action': 'deprovision',
+              'deprovisionReason': deprovisionReasons[reason]
+            }));
+    // If conection is succesful return a DetailedDevice object by parsing the
+    // response body
+    if (response.statusCode == 200) {
+      return true;
+    } else {
+      // If that call was not successful, throw an error.
+      throw ErrorHandler(response.statusCode);
+    }
+  }
+
+  /// Calls the Directory API to disable a device.
+  ///
+  /// * [authToken] is the authentication token for API to accept the requests.
+  /// * [deviceId] is the ID of the device which its detailed information
+  /// will be retrieved.
+  ///
+  /// This method returns true if the action was completed.
+  dynamic disable(String authToken, String deviceId) async {
```

Comment:
Can you remove one of the disable/reenable function and instead just have a changeState an dyou pass the actionString as a parameter?

Reply:
Done.

Reply:
Make this return "Future&lt;bool&gt;" rather than dynamic?  Looks like you use a boolean inside the function itself.

Reply:
Done.